lib/munkres_ru.rb

```diff
@@ -21,17 +21,20 @@ def to_a
     end
   end
 
-  attach_function :solve_munkres, [:size_t, :pointer], ResultArray.by_value
+  attach_function :solve_munkres, [:size_t, :pointer], ResultArray
+  attach_function :free_munkres, [ :pointer ], :void
 
   def self.solve(array)
     flattened = array.flatten
     pointer = FFI::MemoryPointer.new :double, flattened.size
     pointer.autorelease = false # Rust will take ownership of that memory
```

Comment:
So does Rust actually do the right thing with the arg, or do we need to free this from Ruby after the call to `solve_munkres()`?

Reply:
@mjkillough good point.
It seems like it drops the memory correctly (at least on my machine), but I suppose it would be better to give this responsibility back to ruby.